File: src/main/java/fromjavatokotlin/repository/RobotMongoRepo.java
package fromjavatokotlin.repository;

import fromjavatokotlin.model.Robot;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.List;

public interface RobotMongoRepo extends MongoRepository<Robot, String> {

  List<Robot> findAllByAvailabilityIsTrue();

  default List<Robot> findOnlyAvailable() {
    return findAllByAvailabilityIsTrue();
  }
}

File: src/main/kotlin/fromjavatokotlin/repository/RobotMongoRepo.kt
package fromjavatokotlin.repository

import fromjavatokotlin.model.Robot
import org.springframework.data.mongodb.repository.MongoRepository

interface RobotMongoRepo : MongoRepository<Robot, String> {
    fun findAllByAvailabilityIsTrue(): List<Robot>
    @JvmDefault
    fun findOnlyAvailable(): List<Robot> {
        return findAllByAvailabilityIsTrue()
    }
}